### ISSUE TITLE
Are there best practices for features in recycler view?

### ISSUE THREAD

**Issue by Subuday** [badoo/MVICore#126]
1. Should be separate instance feature for each view in recycler view?
2. Should I create separate feature for using in recycler view?
3. Any examples?

**Comment by ShikaSD** [comment]
Hey, it depends on granularity of your views.
Usually we handle all the items through one feature, which outputs state with all items. However, we do have many cases when items from several features are mixed together for one view.

Generally, try to follow business logic rather than matching features to views.
For example, if you have:
 - screen with simple (text, image, maybe click listener) list to items to show - one feature per that screen is fine
 - screen with more complex logic in the list (many actions per item or on list) - you can separate that in the separate feature
 - complex logic for some particular items (e.g. play audio / video) - you can create a separate feature to handle that and combine its state with other items through combine latest

**TLDR**: it depends on your particular case. Try to avoid overcomplicating things in one place, so if it doesn't fit, you can maybe split.
